api/resolver/v1/resolver_api.proto

```diff
@@ -34,6 +35,29 @@ service ResolverAPI {
     };
     option (clutch.api.v1.action).type = READ;
   }
+  rpc AutoComplete(AutocompleteRequest) returns (AutocompleteResponse) {
+    option (google.api.http) = {
+      post : "/v1/resolver/autocomplete"
+      body : "*"
+    };
+    option (clutch.api.v1.action).type = READ;
+  }
+}
+
+message AutocompleteRequest {
+  // The type URL of the desired result.
+  string want = 1 [ (validate.rules).string = {min_bytes : 1} ];
+  string search = 2 [ (validate.rules).string = {min_bytes : 1} ];
+  uint64 result_limit = 3;
+}
+
+message AutocompleteResponse {
+  message AutocompleteResult {
+    string id = 1;
+    map<string, google.protobuf.Value> metadata = 2;
+  }
```

Comment:
Actually thinking about this more as I implemented the frontend, it might just be better to have a simpler approach like the one below. 

```proto
  message AutocompleteResult {
    string id = 1;
    string label = 2;
  }
```

This will just be easier to map to the text-field autocomplete implementation, as we can render the option like so.
```typescript
renderOption={option => (
  <>
    <span>{option.id}</span>
    {option.label}
  </>
)}
```